Assets/GRPC.NET/Scripts/GRPCBestHttpHandler.cs

```diff
@@ -149,6 +149,30 @@ protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage grpcRe
                 return true;
             };
 
+            bestRequest.Callback += (request, response) =>
+            {
+                // If response IS NOT completed yet there was an exception in the request
+                if (!grpcResponseTask.Task.IsCompleted)
+                {
+                    grpcResponseTask.SetException(request?.Exception);
+
+                // If response IS completed but there was an exception (e.g. disconnect from server) we throw
+                // the exception in the blocking read call of the PushPullStream to signal grpc
+                // https://github.com/grpc/grpc-dotnet/blob/master/src/Grpc.Net.Client/Internal/GrpcCall.NonGeneric.cs#L107
+                // https://github.com/grpc/grpc-dotnet/blob/master/src/Grpc.Net.Client/Internal/GrpcProtocolHelpers.cs#L354
+                } else if (response == null && request?.Exception != null)
+                {
+                    incomingDataStream.CloseWithException(request.Exception);
+                }
+
+                // But in any case, if the request is completed, we always want to close the stream
+                // to signal grpc we are definitely done with this stream.
+                else
+                {
+                    incomingDataStream.Close();
```

Comment:
Should this `Close` be in an `else` branch considering the comment says we always want to close the stream and it will not be closed in the first `if` branch as it is?

Reply:
In the exeptional case grpc will actually never have started the reading from this stream so it's not really relevant but yes the comment is a bit misleading.